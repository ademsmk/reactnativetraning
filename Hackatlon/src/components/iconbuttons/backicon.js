import 'react-native';
import React from 'react';
import {View} from 'react-native';
import {TouchableOpacity, Image} from 'react-native';

export function BackIconButton() {
  return (
    <View style={{paddingLeft: 20, flex: 1}}>
      <TouchableOpacity>
        <Image
          style={{height: 30, width: 30}}
          source={require('../../../asset/ok.png')}></Image>
      </TouchableOpacity>
    </View>
  );
}
