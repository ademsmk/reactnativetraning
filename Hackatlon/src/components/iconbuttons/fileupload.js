import 'react-native';
import React from 'react';
import {View} from 'react-native';
import {TouchableOpacity, Image} from 'react-native';

export function FileUploadIconButton() {
  return (
    <TouchableOpacity>
      <Image
        style={{height: 30, width: 30, marginLeft: 20}}
        source={require('../../../asset/upload.png')}></Image>
    </TouchableOpacity>
  );
}
