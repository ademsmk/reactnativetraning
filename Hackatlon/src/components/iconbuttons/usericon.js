import 'react-native';
import React from 'react';
import {View} from 'react-native';
import {TouchableOpacity, Image} from 'react-native';

export function UserIconButton() {
  return (
    <TouchableOpacity>
      <Image
        style={{height: 30, width: 30, marginLeft: 20}}
        source={require('../../../asset/user.png')}></Image>
    </TouchableOpacity>
  );
}
