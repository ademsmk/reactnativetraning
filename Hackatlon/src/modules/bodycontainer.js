import 'react-native';
import React from 'react';
import {View} from 'react-native';
import {TouchableOpacity, Image} from 'react-native';

export function BodyContainer() {
  return <View style={{flex: 6}}></View>;
}
