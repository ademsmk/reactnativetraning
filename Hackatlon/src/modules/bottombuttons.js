import React from 'react';
import {View, StyleSheet, Text, Image} from 'react-native';

export function BottomButtons() {
  return (
    <View
      style={{
        alignItems: 'stretch',
        flex: 1,
        flexDirection: 'row',
        marginBottom: 50,
      }}>
      <View
        style={{
          flex: 1,
          margin: 20,
          marginTop: 50,
        }}>
        <View
          accessible={true}
          focusable={false}
          isTVSelectable={true}
          onClick={() => {}}
          onResponderGrant={() => {}}
          onResponderMove={() => {}}
          onResponderRelease={() => {}}
          onResponderTerminate={() => {}}
          onResponderTerminationRequest={() => {}}
          onStartShouldSetResponder={() => {}}
          style={{
            opacity: 1,
          }}>
          <View
            style={{
              alignContent: 'center',
              alignItems: 'center',
              backgroundColor: 'white',
              borderWidth: 1,
              flexDirection: 'row',
              height: 50,
              justifyContent: 'center',
            }}>
            <Image
              source={{
                testUri: '../../../asset/arrow.png',
              }}
              style={{
                height: 20,
                marginRight: 5,
                width: 20,
              }}
            />
            <Text
              style={{
                color: 'black',
                fontFamily: 'Arial',
                fontSize: 25,
                fontWeight: '400',
                textAlign: 'center',
                textAlignVertical: 'center',
              }}>
              Start
            </Text>
          </View>
        </View>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  image: {height: 30, width: 30, marginLeft: 20},
});
