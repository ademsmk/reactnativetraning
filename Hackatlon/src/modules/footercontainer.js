import React from 'react';
import {View, Text, Image} from 'react-native';

export function FooterContainer() {
  return (
    <View
      style={{
        flex: 5,
      }}>
      <View
        style={{
          height: 30,
          paddingBottom: 5,
          paddingLeft: 20,
        }}>
        <Text
          style={{
            color: 'white',
            fontFamily: 'Arial',
            fontSize: 20,
            fontWeight: '600',
          }}>
          Intensity...
        </Text>
      </View>
      <View
        style={{
          height: 30,
          paddingBottom: 5,
          paddingLeft: 20,
        }}>
        <Text
          style={{
            color: 'white',
            fontFamily: 'Arial',
            fontSize: 20,
            fontWeight: '600',
          }}>
          26 min | get In Shape
        </Text>
      </View>
      <View
        style={{
          height: 60,
          paddingBottom: 5,
          paddingLeft: 20,
          paddingTop: 10,
        }}>
        <Text
          style={{
            color: 'white',
            fontFamily: 'Arial',
            fontSize: 35,
            fontWeight: '600',
          }}>
          Full Body Flow
        </Text>
      </View>
      <View
        style={{
          alignItems: 'stretch',
          flex: 1,
          flexDirection: 'row',
          marginBottom: 50,
        }}>
        <View
          style={{
            flex: 1,
            margin: 20,
            marginTop: 50,
          }}>
          <View
            accessible={true}
            focusable={false}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={{
              opacity: 1,
            }}>
            <View
              style={{
                alignContent: 'center',
                alignItems: 'center',
                backgroundColor: 'white',
                borderWidth: 1,
                flexDirection: 'row',
                height: 50,
                justifyContent: 'center',
              }}>
              <Image
                source={{
                  testUri: '../../../asset/arrow.png',
                }}
                style={{
                  height: 20,
                  marginRight: 5,
                  width: 20,
                }}
              />
              <Text
                style={{
                  color: 'black',
                  fontFamily: 'Arial',
                  fontSize: 25,
                  fontWeight: '400',
                  textAlign: 'center',
                  textAlignVertical: 'center',
                }}>
                Start
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  );
}
