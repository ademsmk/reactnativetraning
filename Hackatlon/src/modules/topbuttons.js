import 'react-native';
import React from 'react';
import {View} from 'react-native';
import {TouchableOpacity, Image} from 'react-native';
import {LoveIconButton} from '../components/iconbuttons/loveicon';
import {FileUploadIconButton} from '../components/iconbuttons/fileupload';
import {UserIconButton} from '../components/iconbuttons/usericon';

export function TopButtons() {
  return (
    <View
      style={{
        flex: 5,
        flexDirection: 'row',
        alignContent: 'flex-end',
        justifyContent: 'flex-end',
        marginRight: 40,
      }}>
      <LoveIconButton></LoveIconButton>
      <FileUploadIconButton></FileUploadIconButton>
      <UserIconButton></UserIconButton>
    </View>
  );
}
