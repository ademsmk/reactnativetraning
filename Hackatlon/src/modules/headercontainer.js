import 'react-native';
import React from 'react';
import {View} from 'react-native';
import {TouchableOpacity, Image} from 'react-native';
import {BackIconButton} from '../components/iconbuttons/backicon';
import {TopButtons} from './topbuttons';

export function HeaderContainer() {
  return (
    <View
      style={{
        flex: 1,
        marginTop: 20,
        flexDirection: 'row',
        alignContent: 'center',
        justifyContent: 'center',
      }}>
      <BackIconButton></BackIconButton>
      <TopButtons></TopButtons>
    </View>
  );
}
