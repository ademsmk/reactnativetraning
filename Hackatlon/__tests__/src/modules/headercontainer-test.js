import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import { HeaderContainer } from "../../../src/modules/headercontainer";
import { View  } from 'react-native';
import { BackIconButton } from "../../../src/components/iconbuttons/backicon";
import { TopButtons } from "../../../src/modules/topbuttons";

describe('HeaderContainer -> ./src/modules/headercontainer', () => {
    it(' should render View component as root item', () => {
        const tree = renderer.create(<HeaderContainer />).toJSON();
        expect(tree.type).toEqual("View");  
    }); 
    it(' View (root) component should have style { flex: 1,  marginTop: 20, flexDirection: "row", alignContent: "center", justifyContent: "center",  }', () => {
        const tree = renderer.create(<HeaderContainer />).toJSON();
        expect(tree.props.style).toEqual({ flex: 1,  marginTop: 20, flexDirection: "row", alignContent: "center", justifyContent: "center",  }) 
    }); 
    it(' root View component should have BackIconButton and TopButtons components as only children', () => {
        const tree = renderer.create(<HeaderContainer />).root.findByType(View);
        expect(tree.children[0].children.length).toEqual(2)
        expect(tree.findByType(BackIconButton)).toBeDefined();
        expect(tree.findByType(BackIconButton).parent).toEqual(tree.children[0]);
        expect(tree.findByType(TopButtons)).toBeDefined(); 
        expect(tree.findByType(TopButtons).parent).toEqual(tree.children[0]);
    });
    it(' should match snapshot', () => {
        const tree = renderer.create(<HeaderContainer />).toJSON();
        expect(tree).toMatchSnapshot();
    });  
});
