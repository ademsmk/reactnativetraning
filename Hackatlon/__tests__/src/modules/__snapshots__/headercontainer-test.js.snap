// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderContainer -> ./src/modules/headercontainer  should match snapshot 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginTop": 20,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "paddingLeft": 20,
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../asset/ok.png",
          }
        }
        style={
          Object {
            "height": 30,
            "width": 30,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignContent": "flex-end",
        "flex": 5,
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "marginRight": 40,
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../asset/hearth.png",
          }
        }
        style={
          Object {
            "height": 30,
            "marginLeft": 20,
            "width": 30,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../asset/upload.png",
          }
        }
        style={
          Object {
            "height": 30,
            "marginLeft": 20,
            "width": 30,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../asset/user.png",
          }
        }
        style={
          Object {
            "height": 30,
            "marginLeft": 20,
            "width": 30,
          }
        }
      />
    </View>
  </View>
</View>
`;
